index.js

```diff
@@ -453,6 +453,20 @@ module.exports = function AsyncAppendOnlyLog(filename, opts) {
     })
   }
 
+  function onStreamsDone(cb) {
+    if (self.streams.every((stream) => stream.cursor === since.value)) {
+      return cb()
+    }
+    const interval = setInterval(function checkIfStreamsDone() {
```

Comment:
Not the most elegant solution, but does the job. It's not a deal breaker for compaction to be delayed ~100ms in the future.